### PR TITLE
Updates for new 20231027 orog fixed files

### LINKED ISSUE #867: Updates for new 20231027 orog fix file set
A new `orog` fix file timestamp is being created for the GFS fix file developmental set. Some updates are needed in UFS_UTILS to accommodate changes in this new timestamp set.

1. Update UFS_UTILS to use `orog_ver=20231027` for the new `GlobalLake*` fix files that previously lived in the unversioned `raw/orog` subset. Any other related updates for this. Discussed in https://github.com/NOAA-EMC/global-workflow/issues/1984.
Files in new orog set:
```
[role.glopara@hfe02 20231027]$ pwd
/scratch1/NCEPDEV/global/glopara/fix/orog/20231027
[role.glopara@hfe02 20231027]$ ll | grep GlobalLake
-rw-r--r-- 1 role.glopara global 1866240000 May 31  2018 GlobalLakeDepth_GLDBv2release.dat
-rw-r--r-- 1 role.glopara global       2191 Oct 25 17:59 GlobalLakeDepth_GLDBv2release.txt
-rw-r--r-- 1 role.glopara global       2695 Oct 25 19:02 GlobalLakeDepth_GLDBv3release.txt
-rw-r--r-- 1 role.glopara global  933120000 May 31  2018 GlobalLakeStatus_GLDBv2release.dat
-rw-r--r-- 1 role.glopara global       2845 Oct 25 18:42 GlobalLakeStatus_GLDBv2release.txt
-rw-r--r-- 1 role.glopara global       4185 Oct 25 17:54 GlobalLakeStatus_GLDBv3release.txt
```
2. Update the `oro_data` filenames used by gdas_init utility and anything else in UFS_UTILS (no longer using symlink in fix set). Change from `oro_C###.mx###.tile#.nc` to `C###.mx###_oro_data.tile#.nc`. Discussed in https://github.com/NOAA-EMC/global-workflow/issues/1981. Example in C384:

```
[role.glopara@hfe02 20231027]$ ll C384.mx025/ | grep oro
-rw-r--r-- 1 role.glopara global  13000006 May  4  2022 C384.mx025_oro_data.tile1.nc
-rw-r--r-- 1 role.glopara global  13000006 May  4  2022 C384.mx025_oro_data.tile2.nc
-rw-r--r-- 1 role.glopara global  13000006 May  4  2022 C384.mx025_oro_data.tile3.nc
-rw-r--r-- 1 role.glopara global  13000006 May  4  2022 C384.mx025_oro_data.tile4.nc
-rw-r--r-- 1 role.glopara global  13000006 May  4  2022 C384.mx025_oro_data.tile5.nc
-rw-r--r-- 1 role.glopara global  13000006 May  4  2022 C384.mx025_oro_data.tile6.nc
lrwxrwxrwx 1 role.glopara global        28 Jan  5  2023 oro_C384.mx025.tile1.nc -> C384.mx025_oro_data.tile1.nc
lrwxrwxrwx 1 role.glopara global        28 Jan  5  2023 oro_C384.mx025.tile2.nc -> C384.mx025_oro_data.tile2.nc
lrwxrwxrwx 1 role.glopara global        28 Jan  5  2023 oro_C384.mx025.tile3.nc -> C384.mx025_oro_data.tile3.nc
lrwxrwxrwx 1 role.glopara global        28 Jan  5  2023 oro_C384.mx025.tile4.nc -> C384.mx025_oro_data.tile4.nc
lrwxrwxrwx 1 role.glopara global        28 Jan  5  2023 oro_C384.mx025.tile5.nc -> C384.mx025_oro_data.tile5.nc
lrwxrwxrwx 1 role.glopara global        28 Jan  5  2023 oro_C384.mx025.tile6.nc -> C384.mx025_oro_data.tile6.nc
```
Full list of affected files:
```
[role.glopara@hfe02 20230615]$ ll C*/oro_* | grep -v frac                                                                                                          
lrwxrwxrwx 1 role.glopara global 29 Jan  5  2023 C1152.mx025/oro_C1152.mx025.tile1.nc -> C1152.mx025_oro_data.tile1.nc
lrwxrwxrwx 1 role.glopara global 29 Jan  5  2023 C1152.mx025/oro_C1152.mx025.tile2.nc -> C1152.mx025_oro_data.tile2.nc
lrwxrwxrwx 1 role.glopara global 29 Jan  5  2023 C1152.mx025/oro_C1152.mx025.tile3.nc -> C1152.mx025_oro_data.tile3.nc
lrwxrwxrwx 1 role.glopara global 29 Jan  5  2023 C1152.mx025/oro_C1152.mx025.tile4.nc -> C1152.mx025_oro_data.tile4.nc
lrwxrwxrwx 1 role.glopara global 29 Jan  5  2023 C1152.mx025/oro_C1152.mx025.tile5.nc -> C1152.mx025_oro_data.tile5.nc
lrwxrwxrwx 1 role.glopara global 29 Jan  5  2023 C1152.mx025/oro_C1152.mx025.tile6.nc -> C1152.mx025_oro_data.tile6.nc
lrwxrwxrwx 1 role.glopara global 28 Jul 31 16:49 C192.mx025/oro_C192.mx025.tile1.nc -> C192.mx025_oro_data.tile1.nc
lrwxrwxrwx 1 role.glopara global 28 Jul 31 16:49 C192.mx025/oro_C192.mx025.tile2.nc -> C192.mx025_oro_data.tile2.nc
lrwxrwxrwx 1 role.glopara global 28 Jul 31 16:49 C192.mx025/oro_C192.mx025.tile3.nc -> C192.mx025_oro_data.tile3.nc
lrwxrwxrwx 1 role.glopara global 28 Jul 31 16:49 C192.mx025/oro_C192.mx025.tile4.nc -> C192.mx025_oro_data.tile4.nc
lrwxrwxrwx 1 role.glopara global 28 Jul 31 16:49 C192.mx025/oro_C192.mx025.tile5.nc -> C192.mx025_oro_data.tile5.nc
lrwxrwxrwx 1 role.glopara global 28 Jul 31 16:49 C192.mx025/oro_C192.mx025.tile6.nc -> C192.mx025_oro_data.tile6.nc
lrwxrwxrwx 1 role.glopara global 28 Jan  5  2023 C192.mx050/oro_C192.mx050.tile1.nc -> C192.mx050_oro_data.tile1.nc
lrwxrwxrwx 1 role.glopara global 28 Jan  5  2023 C192.mx050/oro_C192.mx050.tile2.nc -> C192.mx050_oro_data.tile2.nc
lrwxrwxrwx 1 role.glopara global 28 Jan  5  2023 C192.mx050/oro_C192.mx050.tile3.nc -> C192.mx050_oro_data.tile3.nc
lrwxrwxrwx 1 role.glopara global 28 Jan  5  2023 C192.mx050/oro_C192.mx050.tile4.nc -> C192.mx050_oro_data.tile4.nc
lrwxrwxrwx 1 role.glopara global 28 Jan  5  2023 C192.mx050/oro_C192.mx050.tile5.nc -> C192.mx050_oro_data.tile5.nc
lrwxrwxrwx 1 role.glopara global 28 Jan  5  2023 C192.mx050/oro_C192.mx050.tile6.nc -> C192.mx050_oro_data.tile6.nc
lrwxrwxrwx 1 role.glopara global 28 Jan  5  2023 C384.mx025/oro_C384.mx025.tile1.nc -> C384.mx025_oro_data.tile1.nc
lrwxrwxrwx 1 role.glopara global 28 Jan  5  2023 C384.mx025/oro_C384.mx025.tile2.nc -> C384.mx025_oro_data.tile2.nc
lrwxrwxrwx 1 role.glopara global 28 Jan  5  2023 C384.mx025/oro_C384.mx025.tile3.nc -> C384.mx025_oro_data.tile3.nc
lrwxrwxrwx 1 role.glopara global 28 Jan  5  2023 C384.mx025/oro_C384.mx025.tile4.nc -> C384.mx025_oro_data.tile4.nc
lrwxrwxrwx 1 role.glopara global 28 Jan  5  2023 C384.mx025/oro_C384.mx025.tile5.nc -> C384.mx025_oro_data.tile5.nc
lrwxrwxrwx 1 role.glopara global 28 Jan  5  2023 C384.mx025/oro_C384.mx025.tile6.nc -> C384.mx025_oro_data.tile6.nc
lrwxrwxrwx 1 role.glopara global 27 Jan  5  2023 C48.mx500/oro_C48.mx500.tile1.nc -> C48.mx500_oro_data.tile1.nc
lrwxrwxrwx 1 role.glopara global 27 Jan  5  2023 C48.mx500/oro_C48.mx500.tile2.nc -> C48.mx500_oro_data.tile2.nc
lrwxrwxrwx 1 role.glopara global 27 Jan  5  2023 C48.mx500/oro_C48.mx500.tile3.nc -> C48.mx500_oro_data.tile3.nc
lrwxrwxrwx 1 role.glopara global 27 Jan  5  2023 C48.mx500/oro_C48.mx500.tile4.nc -> C48.mx500_oro_data.tile4.nc
lrwxrwxrwx 1 role.glopara global 27 Jan  5  2023 C48.mx500/oro_C48.mx500.tile5.nc -> C48.mx500_oro_data.tile5.nc
lrwxrwxrwx 1 role.glopara global 27 Jan  5  2023 C48.mx500/oro_C48.mx500.tile6.nc -> C48.mx500_oro_data.tile6.nc
lrwxrwxrwx 1 role.glopara global 28 Jan  5  2023 C768.mx025/oro_C768.mx025.tile1.nc -> C768.mx025_oro_data.tile1.nc
lrwxrwxrwx 1 role.glopara global 28 Jan  5  2023 C768.mx025/oro_C768.mx025.tile2.nc -> C768.mx025_oro_data.tile2.nc
lrwxrwxrwx 1 role.glopara global 28 Jan  5  2023 C768.mx025/oro_C768.mx025.tile3.nc -> C768.mx025_oro_data.tile3.nc
lrwxrwxrwx 1 role.glopara global 28 Jan  5  2023 C768.mx025/oro_C768.mx025.tile4.nc -> C768.mx025_oro_data.tile4.nc
lrwxrwxrwx 1 role.glopara global 28 Jan  5  2023 C768.mx025/oro_C768.mx025.tile5.nc -> C768.mx025_oro_data.tile5.nc
lrwxrwxrwx 1 role.glopara global 28 Jan  5  2023 C768.mx025/oro_C768.mx025.tile6.nc -> C768.mx025_oro_data.tile6.nc
lrwxrwxrwx 1 role.glopara global 27 Jan  5  2023 C96.mx100/oro_C96.mx100.tile1.nc -> C96.mx100_oro_data.tile1.nc
lrwxrwxrwx 1 role.glopara global 27 Jan  5  2023 C96.mx100/oro_C96.mx100.tile2.nc -> C96.mx100_oro_data.tile2.nc
lrwxrwxrwx 1 role.glopara global 27 Jan  5  2023 C96.mx100/oro_C96.mx100.tile3.nc -> C96.mx100_oro_data.tile3.nc
lrwxrwxrwx 1 role.glopara global 27 Jan  5  2023 C96.mx100/oro_C96.mx100.tile4.nc -> C96.mx100_oro_data.tile4.nc
lrwxrwxrwx 1 role.glopara global 27 Jan  5  2023 C96.mx100/oro_C96.mx100.tile5.nc -> C96.mx100_oro_data.tile5.nc
lrwxrwxrwx 1 role.glopara global 27 Jan  5  2023 C96.mx100/oro_C96.mx100.tile6.nc -> C96.mx100_oro_data.tile6.nc
lrwxrwxrwx 1 role.glopara global 27 May  9 16:48 C96.mx500/oro_C96.mx500.tile1.nc -> C96.mx500_oro_data.tile1.nc
lrwxrwxrwx 1 role.glopara global 27 May  9 16:48 C96.mx500/oro_C96.mx500.tile2.nc -> C96.mx500_oro_data.tile2.nc
lrwxrwxrwx 1 role.glopara global 27 May  9 16:48 C96.mx500/oro_C96.mx500.tile3.nc -> C96.mx500_oro_data.tile3.nc
lrwxrwxrwx 1 role.glopara global 27 May  9 16:48 C96.mx500/oro_C96.mx500.tile4.nc -> C96.mx500_oro_data.tile4.nc
lrwxrwxrwx 1 role.glopara global 27 May  9 16:48 C96.mx500/oro_C96.mx500.tile5.nc -> C96.mx500_oro_data.tile5.nc
lrwxrwxrwx 1 role.glopara global 27 May  9 16:48 C96.mx500/oro_C96.mx500.tile6.nc -> C96.mx500_oro_data.tile6.nc
```

### PR DESCRIPTION
**Pull request by GeorgeGayno-NOAA**
## DESCRIPTION OF CHANGES: 
Updates required for the new orography fixed files.

## TESTS CONDUCTED: 
If there are changes to the build or source code, the tests below must be conducted. Contact a repository manager if you need assistance.

- [x] Compile branch on all Tier 1 machines using Intel (Orion, Hercules, Jet, Hera and WCOSS2). Done using 7c74c2c.
- [x] Compile branch on Hera using GNU. Done using 7c74c2c.
- [x] Compile branch in 'Debug' mode on WCOSS2. Done using 7c74c2c.
- [x] Run unit tests locally on any Tier 1 machine. Done on Hera using 7c74c2c.
- [x] Run relevant consistency tests locally on all Tier 1 machine. For this issue, that includes `chgres_cube`, `grid_gen` and `global_cycle`. Done using 7c74c2c. All tests passed as expected.

Describe any additional tests performed (using 7c74c2c):
- On Hera, the `gdas_init` utility was tested using recent GDAS data and the 'orog' files in the 20231027 directory (several model resolutions were tested).
- On Hera, the `sfc_climo_gen` utility was tested using global and regional 'orog' files in the 20231027 directory.

## DEPENDENCIES:
https://github.com/NOAA-EMC/global-workflow/issues/1981
https://github.com/NOAA-EMC/global-workflow/issues/1984

## DOCUMENTATION:
Doxygen compiled with no issues on Hera using 7c74c2c.

## ISSUE: 
Fixes #867.

### CODE REVIEW

**Comment by GeorgeGayno-NOAA** [comment]
@sanatcumar - your review is welcome.

File: ush/global_cycle_driver.sh [hunk 1]
```diff
@@ -9,6 +9,7 @@ set -eux
 #-------------------------------------------------------------------------------------------------
 
 export CASE=${CASE:-C768}                    # resolution of tile: 48, 96, 192, 384, 768, 1152, 3072
+ocn=${ocn:-""}                               # ocean grid resolution: 025, 100, etc.
```
Comment (sanatcumar) [inline ush/global_cycle_driver.sh]: how about 025, 050,100 and 500 instead of 025, 100, etc

Comment (GeorgeGayno-NOAA) [reply ush/global_cycle_driver.sh]: Done.

**Review by KateFriedman-NOAA — APPROVED** [review]
Update looks good! There is an open question about whether we should still have and use the non-mx versions of the oro_data fix files. This update assumes we will use the non-mx versions when ocean resolution is not set. If the decision is made to always use a mx version then a couple scripts in this PR will need updating to remove the non-mx blocks. PR is good otherwise!

**Comment by aerorahul** [comment]
@GeorgeGayno-NOAA 
The changes look good to me.  I have similar questions as @KateFriedman-NOAA.

**Review by aerorahul — APPROVED** [review]
changes look good to me.  @KateFriedman-NOAA raises excellent questions

**Comment by GeorgeGayno-NOAA** [comment]
> Update looks good! There is an open question about whether we should still have and use the non-mx versions of the oro_data fix files. This update assumes we will use the non-mx versions when ocean resolution is not set. If the decision is made to always use a mx version then a couple scripts in this PR will need updating to remove the non-mx blocks. PR is good otherwise!

Thanks. Will await the official decision about whether to retain or remove the non-mx versions before merging.

**Comment by KateFriedman-NOAA** [comment]
> Will await the official decision about whether to retain or remove the non-mx versions before merging.

@GeorgeGayno-NOAA The non-mx versions have been removed from the new `orog/20231027` version. We'll be using the mx versions all the time now. In the workflow we'll use this case block to set `OCNRES`: https://github.com/NOAA-EMC/global-workflow/blob/develop/parm/config/gfs/config.base.emc.dyn#L169:L189

Not sure how you'd like to handle defining `OCNRES` in UFS_UTILS.

**Comment by GeorgeGayno-NOAA** [comment]
> > Will await the official decision about whether to retain or remove the non-mx versions before merging.
> 
> @GeorgeGayno-NOAA The non-mx versions have been removed from the new `orog/20231027` version. We'll be using the mx versions all the time now. In the workflow we'll use this case block to set `OCNRES`: https://github.com/NOAA-EMC/global-workflow/blob/develop/parm/config/gfs/config.base.emc.dyn#L169:L189
> 
> Not sure how you'd like to handle defining `OCNRES` in UFS_UTILS.

For consistency, I can enforce the same combinations of ATM/OCN resolutions in `gdas_init`. Is that utility called from the workflow? Should I use the same variable name - OCNRES?

**Comment by aerorahul** [comment]
@GeorgeGayno-NOAA 
We don't call gdas_init from the workflow (yet).  But that would be great.

**Comment by KateFriedman-NOAA** [comment]
> Should I use the same variable name - OCNRES?

Yes please, for consistency and the future. Thanks @GeorgeGayno-NOAA !

**Comment by GeorgeGayno-NOAA** [comment]
> > Should I use the same variable name - OCNRES?
> 
> Yes please, for consistency and the future. Thanks @GeorgeGayno-NOAA !

See 7c74c2c.

**Comment by KateFriedman-NOAA** [comment]
> See [7c74c2c](https://github.com/ufs-community/UFS_UTILS/commit/7c74c2cd1f7d9c81d8252758d4be81d6c485874a).

Looks great, thanks @GeorgeGayno-NOAA !

**Pull request merged**